Sprint log one 43 (#62)
SW709E19/Report: report/SprintLogs/SprintLog1.tex

\chapter{Sprint 1} \label{cha:sprint-log-1}

This sprint was carried out from 14-10-2019 to 28-10-2019.

\section{Sprint Objective}
The sprint goal for this sprint was to create a walking skeleton (prototype) of the solution, in order to identify and handle issues and risks early on in the project.

> In order for the skeleton to depict the entire solution it should be able to read measurements from a sensor, transfer the data to a gateway which forwards it to a cloud environment, and lastly be able to access the sensor data remotely in an event log.

Reviewer: Jeg forstår ikke helt denne sætning.

Reviewer 2: Det er svært for mig at ændre, når jeg ikke ved hvad problemet er :D Kan du være mere specifik?

> \chapter{Sprint 1}

Reviewer 2: ```suggestion
\chapter{Sprint 1} \label{cha:sprint-log-1}
```

The third user story was to get vibration measurements, which was completed using an accelerometer.

The fourth and last user story was to have an event log for the data. The first accept criteria is completed and the second accept criteria is likely completed as well.

The goal for the sprint was to create a Walking Skeleton of the system, which we managed to do. The skeleton reads measurements from the sensor, sends the measurements to the gateway, which in turn forwards them to the Cloud where they get presented in an event log.
For more info on the Walking Skeleton and an image of the setup see section \todo[color=pink]{Link to the Walking Skeleton section once branches are merged}

\section{Sprint Retrospective}

\textbf{What went well?}

We got the Skeleton to fully work within a single sprint.

Our supervisor gave us really good feedback on our report.

Board management in Github went somewhat better. We should maybe consider Azure Dev Ops though.

There were no issues with getting access to Google Cloud.

\textbf{Follow up on things that could be improved in sprint 0}

> Board management: It has gotten slightly better but it is still not good enough.

Reviewer 3: Why is it not good enough

> Deadlines for tasks: It's been good for tasks that were connected as it improved cooperation. We could still get better at making deadlines during the Scrum meeting though.

Reviewer 3: Sproget virker lidt for uformel

Assigning people to reviews: It worked well. Things have been reviewed faster, and making a notice on Slack about it also worked well.

> Structure of meetings (Sticking to the plan): It has improved. We are, however, still slightly drifting away from the topic at hand, but it has often still been somewhat relevant.

Reviewer 3: ```suggestion
Structure of meetings (Sticking to the plan): It has improved. We still sometimes slightly drift away from the topic at hand, the drifts are however still mostly relevant.
```

> Communication: Has not improved.

Reviewer 3: What is lacking?

Not getting enough details about the current progress on a task during the daily Scrum: This has slightly improved. 

\textbf{What could and is going to be improved?}

There have been some issues with communication.
The group has not been good at answering questions that needed answers. Usually, because you see the message whilst busy, and then forget about it later once the member has the time to do it. Increased usage of mentions (@name) could improve this.
Meeting physically more often could also help out.
Sometimes the daily Scrum meeting takes too long, although it's usually kept within 15 minutes.

There have been some issues with Scrum. we usually have the meeting at 12:00, but it does not fit that well into the schedule, so it has been moved to 11:30 instead as a test.
Group members sometimes forget about the meeting. The solution to this is to have them set an alarm.

Board management is still not that good.
Some user stories have been very hard to conclude.
Group members need to be more attentive as to move their tasks around correctly.
We have created a "done" column to keep track of the user stories that have been completed.

> We do not have enough sprint log notes, to improve on this. People need to write documentation/notes to the work that they are currently doing.

Reviewer 3: Ustruktureret og fortæller ikke hvad vi har tænkt os at forbedre.
